File: pass_by_ref_sample_code/try-parse.cpp
// try-parse.cpp

#include <iostream> 
#include <string>
using namespace std;

bool tryParse(string s, int &n) 
{
    try {
        n = stoi(s, nullptr, 10);
        return true;
    }
    catch(exception e) {
        return false;
    }
}

int main()
{
    string values[] = {"", "160519", "9432.0", "16,667",
                            "   -322   ", "+4302", "(100);", "01FA" };

    for (int i = 0; i < 8; i++) {
        string value = values[i];
        int number;

        if (tryParse(value, number))
            cout << "Converted " << value << " to " << number << endl;
        else
            cout << "Attempted conversion of " << value << " failed" << endl;
    }

    return 0;
}

File: pass_by_ref_sample_code/my_code/try_parse.py
def try_parse(string, base=10):
    try:
        return int(string, base=base)
    except ValueError:
        return None


for value in ("", "23452345", "4356.0", "345,345", "    -234", "+345", "(1234)", "01FA"):
    if number := try_parse(value):
        print(f"Converted {value=} to {number=}")
    else:
        print(f"Attempted conversion of {value=} failed.")

